ghcide/src/Development/IDE/Plugin/TypeLenses.hs

```diff
@@ -241,37 +234,37 @@ rules :: Rules ()
 rules = do
   define $ \GetGlobalBindingTypeSigs nfp -> do
     tmr <- use TypeCheck nfp
-    -- we need session here for tidying types
-    hsc <- use GhcSession nfp
-    result <- liftIO $ gblBindingType (hscEnv <$> hsc) (tmrTypechecked <$> tmr)
+    result <- liftIO $ gblBindingType (ms_hspp_opts . tmrModSummary <$> tmr) (tmrTypechecked <$> tmr)
     pure ([], result)
 
 parseCustomConfig :: A.Object -> Maybe Mode
 parseCustomConfig = A.parseMaybe (A..: "mode")
 
-gblBindingType :: Maybe HscEnv -> Maybe TcGblEnv -> IO (Maybe GlobalBindingTypeSigsResult)
-gblBindingType (Just hsc) (Just gblEnv) = do
+gblBindingType :: Maybe DynFlags -> Maybe TcGblEnv -> IO (Maybe GlobalBindingTypeSigsResult)
+gblBindingType (Just dflags) (Just gblEnv) = do
   let exports = availsToNameSet $ tcg_exports gblEnv
       sigs = tcg_sigs gblEnv
       binds = collectHsBindsBinders $ tcg_binds gblEnv
       patSyns = tcg_patsyns gblEnv
-      dflags = hsc_dflags hsc
       rdrEnv = tcg_rdr_env gblEnv
       showDoc = showDocRdrEnv dflags rdrEnv
-      hasSig :: (Monad m) => Name -> m a -> m (Maybe a)
-      hasSig name f = whenMaybe (name `elemNameSet` sigs) f
-      bindToSig id = do
-        let name = idName id
-        hasSig name $ do
-          env <- tcInitTidyEnv
-          let (_, ty) = tidyOpenType env (idType id)
-          pure $ GlobalBindingTypeSig name (printName name <> " :: " <> showDoc (pprSigmaType ty)) (name `elemNameSet` exports)
-      patToSig p = do
-        let name = patSynName p
+      hasSig name = name `elemNameSet` sigs
+      isExported name = name `elemNameSet` exports
+      prettyPrintTy ty = showDoc (pprSigmaType ty)
+      bindToSig id
+        | name <- idName id
+          , hasSig name
+          , env <- emptyTidyEnv
+          , (_, ty) <- tidyOpenType env (idType id) =
```

Comment:
`tidyTopType` maybe?

Reply:
Actually I stole this from here: https://github.com/ghc/ghc/blob/302854154626ef10363afdda3ff1db7160e0827f/compiler/GHC/Rename/Names.hs#L1507
Can we use the `Name` of binding to lookup `tcg_type_env`, rather than tidying the type obtained from `Id`?

Reply:
@wz1000 hi, maybe you missed @berberman question?

Reply:
I think both would be the same.